llarp/util/encode.hpp

```diff
@@ -150,6 +150,20 @@ namespace llarp
     return sz == 0;
   }
 
+  template <typename InputIt, typename OutputIt>
+  bool
+  HexDecode(InputIt begin, InputIt end, OutputIt target, size_t sz)
+  {
+    while (begin != end && sz)
+    {
+      const auto first = *begin++;
+      const auto second = *begin++;
+      *(target++) = char2int(first) * 16 + char2int(second);
+      --sz;
+    }
+    return sz == 0;
+  }
+
```

Comment:
The vast majority of this file can be replaced with lokimq's hex.h/base32z.h/base64.h (which use constexpr lookup tables and probably end up faster -- especially for base32z decoding which is currently using an unordered_map here).

Reply:
The size-limited one here isn't directly replaceable, but can be wrapped pretty easily:
```C++
template <typename InputIt, typename OutputIt>
HexDecode(InputIt begin, InputIt end, OutputIt target, size_t sz)
{
  if (std::distance(begin, end) != 2*sz) return false;
  lokimq::to_hex(begin, end, target);
  return true;
}
```

Reply:
Done (via sub-PR into this PR).